vispy/visuals/pseudo_spheres.py

```diff
@@ -0,0 +1,204 @@
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# Copyright (c) Vispy Development Team. All Rights Reserved.
+# Distributed under the (new) BSD License. See LICENSE.txt for more info.
+# -----------------------------------------------------------------------------
+"""Pseudo spheres (shaded discs) visual"""
+
+import numpy as np
+
+from ..color import ColorArray
+from ..gloo import VertexBuffer
+from .shaders import Function, Variable
+from .visual import Visual
+
+
+vert = """
+uniform float u_px_scale;
+uniform float u_scale;
+uniform vec3 u_light_position;
+
+attribute vec3 a_position;
+attribute vec4 a_color;
+attribute float a_radius;
+
+varying vec4 v_color;
+varying float v_radius;
+varying vec3 v_light_direction;
+varying float v_depth;
+varying float v_depth_middle;
+
+void main (void) {
+    v_color = a_color;
+    v_radius = a_radius * u_scale * u_px_scale;
+    v_light_direction = normalize(u_light_position);
+
+    vec4 pos = vec4(a_position, 1.0);
+    vec4 fb_pos = $visual_to_framebuffer(pos);
+    gl_Position = $framebuffer_to_render(fb_pos);
+
+    // Get the framebuffer z direction relative to this sphere in visual coords
+    vec4 z = $framebuffer_to_visual(fb_pos + vec4(0, 0, -100, 0));
+    z = (z/z.w - pos) / 100;
+
+    // Get the depth of the sphere in its middle (+ radius)
+    vec4 depth_z = $framebuffer_to_render($visual_to_framebuffer(pos + normalize(z) * a_radius));
```

Comment:
I _think_ this should be in the `framebuffer` coordinate system since you are using `gl_FragCoord` to set the depth below (see docs below - `gl_FragCoord` uses the framebuffer coordinate system).

https://github.com/vispy/vispy/blob/d901f6430784416bacb028e1ff450a3640400e44/vispy/visuals/transforms/transform_system.py#L42-L53

Reply:
Correct, solved!